site/source/tutorials/1020_registries/index.md

```diff
@@ -0,0 +1,130 @@
+
+{% aside 'Registries Everywhere!' %}
+A `registry` can also be used to define widgets with non visual responsibilities, such as state injection and routing. To learn more, take a look at the [container tutorial](../comingsoon.html) and [routing tutorial](../comingsoon.html) .
+{% endaside %}
+
+At the moment we haven't affected the application, however we now have a handle to a `registry` that we can start to define widgets. Once the widgets are defined in the `registry` they will be available through the application and can be used by switching the concrete class in `w()` with the `registry` label.
+
+{% instruction 'Add the widget imports to `main.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:6-10 %}
+
+{% instruction 'Then define widgets in the `registry`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:18-23 %}
+
+In the next section we'll using the registry label in our render functions.
+
+{% section %}
+
+## Using Registry Items
+
+Now that the `registry` has been made available to the application and the widgets have been defined, we can use the label instead of the widget classes across the application.
+
+{% instruction 'Use registry labels in `App.ts`\'s render function.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:46-58 %}
+
+{% instruction 'Use registry labels in `WorkerContainer.ts`\'s render function.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerContainer.ts' lines:21-24 %}
+
+{% instruction 'Use registry labels in `WorkerForm.ts`\'s render function.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:43-86 %}
+
+Notice that we are passing a generic type to the `w()` function call, this is because when using a `registry` label it is unable to infer the properties interface. As a result falls back to the default, `WidgetProperties` interface. Passing the generic tells `w()` the type of widget the label represents in the `registry` and will correctly enforce the widget's properties interface.
+
+Next, we'll create a widget that is lazily loaded when needed!
+
+{% section %}
+
+## Lazy Loading Widgets
+
+First we need to extract the `_renderBack` function from `Worker.ts` into a new widget `WorkerBack.ts` and then add the new widget to the registry.
+
+{% task 'Add the following code in `WorkerBack.ts`' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerBack.ts' %}
+
+{% task 'Update the `_renderBack` function to use the `WorkerBack` registry items in `Worker.ts`' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Worker.ts' lines:51-65 %}
+
+{% aside 'Use Before You Define' %}
+The `registry` is designed to mirror the behavior and API of custom elements where ever possible. One neat feature is that a registry item can used before it is defined and, once defined, widgets that use the it will automatically re-render!
```

Comment:
widgets that use the it will automatically re-render -> widgets that use the `registry` will automatically re-render